uisdk/src/main/java/com/karhoo/uisdk/util/extension/ServiceCancellationExt.kt

```diff
@@ -6,27 +6,37 @@ import com.karhoo.sdk.api.model.TripStatus
 import com.karhoo.uisdk.R
 import com.karhoo.uisdk.util.CANCELLATION_BEFORE_DRIVER_EN_ROUTE
 import com.karhoo.uisdk.util.CANCELLATION_TIME_BEFORE_PICKUP
+import com.karhoo.uisdk.util.DateUtil
 
 fun ServiceCancellation.getCancellationText(context: Context): String? {
-    var text: String?
+    var cancellationText: String?
 
     when (this.type) {
         CANCELLATION_TIME_BEFORE_PICKUP -> {
-            text = String.format(context.getString(R.string.kh_uisdk_quote_cancellation_minutes), minutes)
-
             if (minutes == 0) {
-                text = null
+                cancellationText = null
+            } else {
+                val hours: Int = DateUtil.roundMinutesInHours(minutes)
+                val leftOverMinutes: Int = DateUtil.getLeftOverMinutesFromHours(minutes)
+
+                cancellationText = context.getString(R.string.kh_uisdk_quote_cancellation_before_pickup_start) + " "
+
+                if (hours > 0) cancellationText += context.resources.getQuantityString(R.plurals.kh_uisdk_quote_cancellation_before_pickup_hours, hours, hours) + " "
```

Comment:
You mean moving the SLA cancellation message buildup in another method/class? Seemed like the proper place, not sure we will have use cases for building the message with values from another place rather than the minutes from the actual SLA

Reply:
We could create another method in the extension to output the cancellation message based on input minutes/hours

Reply:
I mean just formatting the full duration string (ex: "1 hour 24 minutes") as it might not be specific just to the cancellation message.

Reply:
Oh got it. Doing it now

Reply:
Done